lib/loadObj.js

```diff
@@ -256,34 +547,19 @@ function loadObj(objPath, options) {
         } else if ((result = uvPattern.exec(line)) !== null) {
             uvs.push(parseFloat(result[1]));
             uvs.push(1.0 - parseFloat(result[2])); // Flip y so 0.0 is the bottom of the image
-        } else if ((result = facePattern1.exec(line)) !== null) {
-            addFace(
-                result[1], result[1], undefined, undefined,
-                result[2], result[2], undefined, undefined,
-                result[3], result[3], undefined, undefined,
-                result[4], result[4], undefined, undefined
-            );
-        } else if ((result = facePattern2.exec(line)) !== null) {
-            addFace(
-                result[1], result[2], result[3], undefined,
-                result[4], result[5], result[6], undefined,
-                result[7], result[8], result[9], undefined,
-                result[10], result[11], result[12], undefined
-            );
-        } else if ((result = facePattern3.exec(line)) !== null) {
-            addFace(
-                result[1], result[2], result[3], result[4],
-                result[5], result[6], result[7], result[8],
-                result[9], result[10], result[11], result[12],
-                result[13], result[14], result[15], result[16]
-            );
-        } else if ((result = facePattern4.exec(line)) !== null) {
-            addFace(
-                result[1], result[2], undefined, result[3],
-                result[4], result[5], undefined, result[6],
-                result[7], result[8], undefined, result[9],
-                result[10], result[11], undefined, result[12]
-            );
+        } else { // face line or invalid line
+            // If we already have buffered data, this is a continued line
+            var continuedLine = faceVertices.length > 0;
+
+            if (facePattern1.test(line) || (continuedLine && facePattern5.test(line))) { // format "f v v v ..."
+                parseFaceLine(line, false, false);
+            } else if (facePattern2.test(line) || (continuedLine && facePattern6.test(line))) { // format "f v/uv v/uv v/uv ..."
+                parseFaceLine(line, true, false);
+            } else if (facePattern3.test(line) || (continuedLine && facePattern7.test(line))) { // format "v/uv/n v/uv/n v/uv/n ..."
+                parseFaceLine(line, true, true);
+            } else if (facePattern4.test(line) || (continuedLine && facePattern8.test(line))) { // format "v//n v//n v//n ..."
+                parseFaceLine(line, false, true);
+            }
```

Comment:
After looking at the multi-line parsing a bit, I think it could be simplified by buffering the string rather than buffering the parsed data. The face parsing code should take a single string regardless of whether it originally spanned multiple lines or not.

Then with some modifications to the first four regexes and it should be possible to capture all the attributes like before, with `addFace` just handling them in a more generic fashion.